internal/groundstation/gpustats.go

```diff
@@ -7,30 +7,34 @@ import (
 	"github.com/gpuctl/gpuctl/internal/uplink"
 )
 
-func (gs *groundstation) gpustats(data uplink.GpuStatsUpload, req *http.Request, log *slog.Logger) error {
+const (
+	GPUStatsOkayResponse = "OK"
+)
+
+func (gs *groundstation) gpustats(data uplink.GpuStatsUpload, req *http.Request, log *slog.Logger) (string, error) {
 	log.Info("Got GPU stats", "stats", data.Stats)
 
 	// NOTE: just commented this during the big refactor -jyry
 	err := gs.db.UpdateLastSeen(data.Hostname)
 	if err != nil {
-		return err
+		return "", err
 	}
 
 	if len(data.GPUInfos) > 0 {
 		err := gs.handleGPUInfo(data.Hostname, data.GPUInfos)
 		if err != nil {
-			return err
+			return "", err
 		}
 	}
 
 	if len(data.Stats) > 0 {
 		err := gs.handleGPUStatSamples(data.Hostname, data.Stats)
 		if err != nil {
-			return err
+			return "", err
 		}
 	}
 
-	return nil
+	return GPUStatsOkayResponse, nil
```

Comment:
This is in effect returning some type `T = (A, Error)` where `A = Okay | ()` which is just `Maybe Okay` and `Error = String | ()`. Issue is, this expressiveness isn't fully used. The code has the invariant that A is of type `Okay` iff Error is of type `()`, and vice versa. This is another reason this is concerning.